core/src/main/java/io/questdb/griffin/engine/orderby/SortedRecordCursorFactory.java

```diff
@@ -99,16 +99,25 @@ public boolean usesCompiledFilter() {
         return base.usesCompiledFilter();
     }
 
-    static boolean hasDescendingOrder(ListColumnFilter sortColumnFilter) {
+    private static int toOrder(int filter) {
+        if (filter >= 0) {
+            return SCAN_DIRECTION_FORWARD;
+        } else {
+            return SCAN_DIRECTION_BACKWARD;
+        }
+    }
+
+    static int getScanDirection(ListColumnFilter sortColumnFilter) {
         assert sortColumnFilter.size() > 0;
+        int order = toOrder(sortColumnFilter.get(0));
```

Comment:
Can you please explain what this is doing?

I would expect it to compare that the first column the sorting is done is the timestamp column. 

If yes, return the sorting order of the timestamp column.
Otherwise, return OTHER.

Reply:
It was meant to return single direction if all columns were sorted ASC or DESC, otherwise OTHER . 
All call sites seem to check for timestamp index so I simplified the method  in bef590b2 .